extensions/azurePublishNew/src/node/azureResources/botChannel.ts

```diff
@@ -15,29 +36,49 @@ export const botRegistrationDefinition: ResourceDefinition = {
   group: AZURE_HOSTING_GROUP_NAME,
 };
 
-const getBotChannelProvisionMethod = (): ProvisionMethod => (
-  config: BotServiceConfig,
-  workingSet: ProvisionWorkingSet
-): Promise<ProvisionWorkingSet> => {
-  const appRegistrationResult: AppRegistrationResult = workingSet.appRegistration;
-  const webAppResult: WebAppResult = workingSet.webApp;
-  const { appId } = appRegistrationResult.appId;
-  const { endpoint } = webAppResult.endpoint;
-  const { hostname } = webAppResult.hostname;
+const botChannelProvisionMethod = (provisionConfig: ProvisionConfig): ProvisionMethod => {
+  const tokenCredentials = new TokenCredentials(provisionConfig.accessToken);
+  const azureBotService = new AzureBotService(tokenCredentials, provisionConfig.subscriptionId);
 
-  const provisionResult: BotChannelResult = { botName: '' };
+  return async (config: BotServiceConfig, workingSet: ProvisionWorkingSet): Promise<ProvisionWorkingSet> => {
+    const appId = workingSet.appRegistration?.appId;
```

Comment:
Do you want to throw if the working set doesn't contain what this service needs?

Reply:
set up a new function for checking required fields